lib/vrl/stdlib/src/array.rs

```diff
@@ -52,10 +68,7 @@ struct ArrayFn {
 
 impl Expression for ArrayFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
-        match self.value.resolve(ctx)? {
-            v @ Value::Array(_) => Ok(v),
-            v => Err(format!(r#"expected "array", got {}"#, v.kind()).into()),
-        }
+        array(self.value.resolve(ctx)?)
     }
```

Comment:
I can't say I'm a big fan of the split between `Expression::resolve` and `Function::call`. I wonder if there's a way for us to unify the two, keeping the implementation details in function implementations at a minimum.